src/main.cpp

```diff
@@ -4441,10 +4468,14 @@ bool InitBlockIndex(const CChainParams &chainparams)
     LOCK(cs_main);
 
     // Initialize global variables that cannot be constructed at startup.
-    recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
+    {
+        LOCK(cs_recentRejects);
+        recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
+        txn_recently_in_block.reset(new CRollingBloomFilter(16000 /* just a few block's worth */, 0.000001));
```

Comment:
couldn't we parameterize 16000? since it is correlated w/ the max block size wonder if we could a function of it rather than a magic number.

Reply:
yes your'e right, and it should be probably sized something like  ( EB * (avg txns in a block))